www/FileWriter.js

```diff
@@ -107,8 +108,29 @@ FileWriter.prototype.write = function(data) {
         var fileReader = new FileReader();
         fileReader.onload = function() {
             // Call this method again, with the arraybuffer as argument
-            FileWriter.prototype.write.call(that, this.result);
+            FileWriter.prototype.write.call(that, this.result, true /* isPendingBlobReadResult */);
         };
+        fileReader.onerror = function (evt) {
+            // DONE state
+            that.readyState = FileWriter.DONE;
```

Comment:
I don't think so as the `DONE` state is being set in `FileReader.readSuccessCallback`, `FileReader.readFailureCallback` and `FileReader.abort` only.
`abort` should not be called outside (and we don't call it) taking into account that fileReader is an internal variable, `readSuccessCallback` calls `onload` and `readFailureCallback` calls `onerror` so we are covering all the cases.